### ISSUE TITLE
Send by subuser

### ISSUE THREAD

**Issue by btmdave** [sendgrid/sendgrid-nodejs#886]
Is there no way to send from a subuser? Not seeing the authentication methods from `@sendgrid/mail` or within `@sendgrid/client`.  Any points in the right direction would be greatly appreciated.  Thank you.

**Comment by thinkingserious** [comment]
Hello @btmdave,

You will need to add a custom header [like so](https://github.com/sendgrid/sendgrid-nodejs/blob/master/use-cases/custom-headers.md). But with `On-Behalf-Of: subusername` as the `headers` key.

With Best Regards,

Elmer

**Comment by btmdave** [comment]
Perfect, thanks @thinkingserious, will give that a shot.

**Comment by thinkingserious** [comment]
Awesome, please let me know how it goes.

**Comment by aroach** [comment]
You can see an example in our sendgrid-cli project: https://github.com/sendgrid/sendgrid-cli/blob/master/lib/sgclient.js#L46

**Comment by btmdave** [comment]
@thinkingserious @aroach This doesn't actually seem to be working, the mail sent is going through our primary account.  I did find this: https://sendgrid.api-docs.io/v3.0/how-to-use-the-sendgrid-v3-api/on-behalf-of which says that it doesn't work with the `mail.send` api?

This is how I'm sending it:

Mail sends fine, but the reporting is all through the primary account, nothing in the subaccount.

        const msg = {
          to: options.to,
          from: options.from,
          replyTo: options.replyTo,
          subject: options.subject,
          text: results.text,
          html: results.html,
          headers: {
            "on-behalf-of": "SUBACCOUNT_USERNAME"
          }
        };

**Comment by thinkingserious** [comment]
Hello @btmdave,

Have you tried using `Token` rather than `Bearer` for authentication? 

With Best Regards,

Elmer

**Comment by nikolajnyboe** [comment]
Hi,

I have run into the same problem. I'm having trouble sending an email "on-behalf" of a subuser using the "Web API V3" and `@sendgrid/mail`.

I have tried with multiple approaches such as:
```
{
  headers: {
    Authorization: `Token ${process.env.SENDGRID_TOKEN}`,
    "On-Behalf-Of": subuser
  },
  to: email.to,
  from: email.from,
  templateId: email.templateId,
  dynamic_template_data: {
    ...dynamicData,
  },
  attachments: attachments,
}
```

But none of them seem to work. 

The only way I have been able to send an email as a subuser, using a dynamic template created on that subusers account, has been by calling `sendGrid.setApiKey("SUBUSER_API_KEY")` just before sending the email.

Am I missing something? - I can't find any documentation on this topic besides this thread.

Thank you!
